src/vmm/src/vstate/vm/mod.rs

```diff
@@ -39,43 +38,25 @@ pub enum VmError {
     Arch(#[from] ArchVmError),
 }
 
-/// A wrapper around creating and using a VM.
+/// Structure containing all relevant things for a virtual machine
 #[derive(Debug)]
 pub struct Vm {
     fd: VmFd,
-
-    // Arm specific fields.
-    // On aarch64 we need to keep around the fd obtained by creating the VGIC device.
-    #[cfg(target_arch = "aarch64")]
-    irqchip_handle: Option<GICDevice>,
+    arch: ArchVm,
 }
 
 /// Contains Vm functions that are usable across CPU architectures
 impl Vm {
     /// Create a new `Vm` struct.
     pub fn new(kvm: &Kvm) -> Result<Self, VmError> {
-        // Create fd for interacting with kvm-vm specific functions.
-        let vm_fd = kvm.fd.create_vm().map_err(VmError::VmFd)?;
-
-        #[cfg(target_arch = "aarch64")]
-        {
-            Ok(Vm {
-                fd: vm_fd,
-                irqchip_handle: None,
-            })
-        }
+        let fd = kvm.fd.create_vm().map_err(VmError::VmFd)?;
+        let arch = Vm::arch_create()?;
 
-        #[cfg(target_arch = "x86_64")]
-        {
-            Ok(Vm { fd: vm_fd })
-        }
+        Ok(Vm { fd, arch })
     }
 
     /// Initializes the guest memory.
-    pub fn memory_init(
-        &self,
-        guest_mem: &GuestMemoryMmap,
-    ) -> Result<(), VmError> {
+    pub fn memory_init(&self, guest_mem: &GuestMemoryMmap) -> Result<(), VmError> {
```

Comment:
Once `Vm` type is done, these impls should just extend the `ArchVm` imported from either x86 or aarch64 modules.

Reply:
Yup, done